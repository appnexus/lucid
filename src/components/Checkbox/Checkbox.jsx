import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	omitProps,
} from '../../util/component-types';

const cx = lucidClassNames.bind('&-Checkbox');

const { bool, func, object, string } = PropTypes;

/**
 * {"categories": ["controls", "toggles"]}
 *
 * This is a checkbox. It's a component that is in one of two particular states
 * at any given moment. It features a custom visualization of the native
 * checkbox button control to reflect its current state.
 *
 * It uses a hidden native checkbox control under the hood but leverages other
 * HTML elements to visualize its state.
 */
const Checkbox = createReactClass({
	displayName: 'Checkbox',
	propTypes: {
		/**
		 * Appended to the component-specific class names set on the root
		 * element.
		 */
		className: string,

		/**
		 * Indicates whether the component should appear in an "indeterminate"
		 * or "partially checked" state. This prop takes precedence over
		 * `isSelected`.
		 */
		isIndeterminate: bool,

		/**
		 * Indicates whether the component should appear and act disabled by
		 * having a "greyed out" palette and ignoring user interactions.
		 */
		isDisabled: bool,

		/**
		 * Indicates that the component is in the "selected" state when true
		 * and in the "unselected" state when false. This props is ignored if
		 * `isIndeterminate` is `true`.
		 */
		isSelected: bool,

		/**
		 * Called when the user clicks on the component or when they press the
		 * space key while the component is in focus.
		 *
		 * Signature: `(isSelected, { event, props }) => {}`
		 */
		onSelect: func,

		/**
		 * Passed through to the root element.
		 */
		style: object,
	},

	getDefaultProps() {
		return {
			isIndeterminate: false,
			isDisabled: false,
			isSelected: false,
			onSelect: _.noop,
		};
	},

	componentDidMount() {
		this.nativeElement = this.refs.nativeElement;
	},

	handleSpanClick(e) {
		e.preventDefault();
	},

	render() {
		const {
			className,
			isIndeterminate,
			isSelected,
			isDisabled,
			style,
			...passThroughs
		} = this.props;

		return (
			<div
				className={cx(
					'&',
					{
						'&-is-disabled': isDisabled,
						'&-is-selected': isIndeterminate || isSelected,
					},
					className
				)}
				onClick={this.handleClicked}
				style={style}
			>
				<input
					onChange={_.noop}
					{...omitProps(passThroughs, Checkbox, ['children'])}
					checked={isSelected}
					className={cx('&-native')}
					disabled={isDisabled}
					ref="nativeElement"
					type="checkbox"
				/>
				<span
					onClick={this.handleSpanClick}
					className={cx('&-visualization-glow')}
				/>
				<span
					onClick={this.handleSpanClick}
					className={cx('&-visualization-container')}
				/>
				{isIndeterminate
					? <span
							onClick={this.handleSpanClick}
							className={cx('&-visualization-indeterminate')}
						>
							<span className={cx('&-visualization-indeterminate-line')} />
						</span>
					: <span
							onClick={this.handleSpanClick}
							className={cx('&-visualization-checkmark')}
						>
							<span className={cx('&-visualization-checkmark-stem')} />
							<span className={cx('&-visualization-checkmark-kick')} />
						</span>}
			</div>
		);
	},

	handleClicked(event) {
		const { isDisabled, isSelected, onSelect } = this.props;

		if (!isDisabled) {
			onSelect(!isSelected, { event, props: this.props });
			this.nativeElement.focus();
		}
	},
});

export default Checkbox;
