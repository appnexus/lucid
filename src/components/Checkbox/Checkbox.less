@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';
@Checkbox-size: 22px;
@Checkbox-animation-timing: 0.1s linear;
@Checkbox-borderRadius: @size-XXS;
@Checkbox-borderColor: fade(#000, 30%);

@Checkbox-container-size: @Checkbox-size - @size-XS;
@Checkbox-container-offset: @size-XXS;

@Checkbox-glow-offset: @size-XXS;
@Checkbox-glow-scale: scale(1.375);

@Checkbox-disabled-backgroundColor: fade(#000, 5%);
@Checkbox-disabled-borderColor: fade(#000, 10%);

.lucid-Checkbox {
	cursor: pointer;
	display: inline-block;
	height: @Checkbox-size;
	position: relative;
	user-select: none;
	width: @Checkbox-size;

	&-is-disabled {
		cursor: not-allowed;
	}

	&-native {
		opacity: 0;
	}

	&-visualization-container {
		background-color: white;
		border: solid 1px @Checkbox-borderColor;
		border-radius: @Checkbox-borderRadius;
		.box-sizing();
		height: @Checkbox-container-size;
		left: @Checkbox-container-offset;
		position: absolute;
		top: @Checkbox-container-offset;
		width: @Checkbox-container-size;
	}

	&-visualization-indeterminate {
		opacity: 0;
		width: @Checkbox-size;
		height: @Checkbox-size;
		transform: rotate(0deg);
		top: 0;
		left: 0;
		position: absolute;
	}

	&-visualization-indeterminate-line {
		top: 10px;
		left: 7px;
		width: 8px;
		height: 2px;
		background-color: white;
		position: absolute;
	}

	&-visualization-checkmark {
		opacity: 0;
		width: @Checkbox-size;
		height: @Checkbox-size;
		transform: rotate(-45deg);
		top: 0;
		left: 0;
		position: absolute;
	}

	&-visualization-checkmark-stem {
		top: 11px;
		left: 7px;
		width: 9px;
		height: 2px;
		background-color: white;
		position: absolute;
	}

	&-visualization-checkmark-kick {
		top: 8px;
		left: 7px;
		width: 2px;
		height: 4px;
		background-color: white;
		position: absolute;
	}

	&-visualization-glow {
		background-color: @color-primary;
		border-radius: @Checkbox-borderRadius;
		.box-sizing();
		height: @Checkbox-container-size;
		left: @Checkbox-glow-offset;
		opacity: 0;
		position: absolute;
		top: @Checkbox-glow-offset;
		transition: @Checkbox-animation-timing;
		width: @Checkbox-container-size;
	}

	&:hover &-visualization-container,
	&-native:focus ~ &-visualization-container {
		border-color: @color-primary;
		transition: @Checkbox-animation-timing;
	}

	&:hover &-visualization-glow {
		opacity: 0.3;
		transform: @Checkbox-glow-scale;
	}

	&-is-disabled &-visualization-container,
	&-is-disabled:hover &-visualization-container {
		background-color: @Checkbox-disabled-backgroundColor;
		border-color: @Checkbox-disabled-borderColor;
	}

	&-is-disabled:hover &-visualization-glow {
		opacity: 0;
		transform: none;
	}

	&-is-disabled&-is-selected &-visualization-container {
		opacity: 0.3;
	}

	&-is-disabled&-is-selected:hover &-visualization-container,
	&-is-selected &-visualization-container,
	&:active &-visualization-container {
		background-color: @color-primary;
		border-color: @color-primary;
	}

	&-is-disabled:active &-visualization-container {
		background-color: @Checkbox-disabled-backgroundColor;
		border-color: @Checkbox-disabled-borderColor;
	}

	&-is-selected &-visualization-checkmark, &-visualization-indeterminate {
		opacity: 1;
		transition: @Checkbox-animation-timing;
	}

	&-native:focus ~ &-visualization-glow {
		opacity: 0.3;
		transform: @Checkbox-glow-scale;
	}
}
