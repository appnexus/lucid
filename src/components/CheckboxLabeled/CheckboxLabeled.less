@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-CheckboxLabeled-wrapper {
	display: flex;
}

.lucid-CheckboxLabeled {
	align-items: flex-start;
	color: @color-darkGray;
	cursor: pointer;
	display: flex;
	font-size: @fontSize;
	font-family: @fontFamily;
	justify-content: flex-start;

	> .lucid-Checkbox {
		flex: 0 0 @Checkbox-size;
		margin-right: @size-XXS;
	}

	&-label {
		padding-top: @size-XXS;
	}

	&&-is-disabled {
		color: @color-mediumGray;
		cursor: not-allowed;
	}

	&:hover .lucid-Checkbox-visualization-container {
		border-color: @color-primary;
		transition: @Checkbox-animation-timing;
	}

	&:hover .lucid-Checkbox-visualization-glow {
		.opacity(0.3);

		transform: @Checkbox-glow-scale;
	}

	&&-is-disabled .lucid-Checkbox-visualization-container,
	&&-is-disabled:hover .lucid-Checkbox-visualization-container {
		background-color: @Checkbox-disabled-backgroundColor;
		border-color: @Checkbox-disabled-borderColor;
	}

	&&-is-disabled:hover .lucid-Checkbox-visualization-glow {
		.opacity(0);

		transform: none;
	}

	&&-is-disabled&&-is-selected:hover .lucid-Checkbox-visualization-container,
	&&-is-selected .lucid-Checkbox-visualization-container {
		background-color: @color-primary;
		border-color: @color-primary;
	}
}

