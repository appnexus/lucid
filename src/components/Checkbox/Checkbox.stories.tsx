import React, { useState } from 'react';
import { Meta } from '@storybook/react';

import Checkbox from './Checkbox';

export default {
	title: 'Controls/Checkbox',
	component: Checkbox,
	parameters: {
		docs: {
			description: {
				component: Checkbox.peek.description,
			},
		},
	},
} as Meta;

const Template: any = (args) => {
	const [selected, setSelected] = useState(args.isSelected || false);
	return (
		<Checkbox
			{...args}
			isSelected={selected}
			onSelect={() => setSelected(!selected)}
		/>
	);
};

export const Default = Template.bind({});
Default.args = {
	title: 'Default',
};

export const Plain = Template.bind({});
Plain.args = {
	title: 'Plain',
	isDisabled: false,
	isSelected: true,
};

export const DisabledUnselected = Template.bind({});
DisabledUnselected.args = {
	title: 'Disabled Unselected',
	isDisabled: true,
	isSelected: false,
};

export const DisabledSelected = Template.bind({});
DisabledSelected.args = {
	title: 'Disabled Selected',
	isDisabled: true,
	isSelected: true,
};

export const DisabledIndeterminate = Template.bind({});
DisabledIndeterminate.args = {
	title: 'Disabled Selected',
	isIndeterminate: true,
	isDisabled: true,
};
