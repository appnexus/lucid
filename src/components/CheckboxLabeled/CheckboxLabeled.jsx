import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import { createClass, findTypes, omitProps } from '../../util/component-types';
import Checkbox from '../Checkbox/Checkbox';

const cx = lucidClassNames.bind('&-CheckboxLabeled');
const { any, node, object, string } = PropTypes;

/**
 * {"categories": ["controls", "toggles"], "extend": "Checkbox", "madeFrom": ["Checkbox"]}
 *
 * This is a composite of the `Checkbox` component and the native `label`
 * element.
 */
const CheckboxLabeled = createClass({
	displayName: 'CheckboxLabeled',

	components: {
		/*
		 * Renders a `<label>` for the `<Checkbox>`
		 */
		Label: createClass({
			displayName: 'CheckboxLabeled.Label',
			propName: 'Label',
			propTypes: {
				/**
				 * Used to identify the purpose of this checkbox to the user -- can
				 * be any renderable content.
				 */
				children: node,
			},
		}),
	},

	propTypes: {
		...Checkbox.propTypes,

		/**
		 * Appended to the component-specific class names set on the root
		 * element.
		 */
		className: string,

		/**
		 * Passed through to the root element.
		 */
		style: object,

		/**
		 * Child element whose children are used to identify the purpose of this
		 * checkbox to the user.
		 */
		Label: any,
	},

	getDefaultProps() {
		return {
			isIndeterminate: false,
			isDisabled: false,
			isSelected: false,
			onSelect: _.noop,
		};
	},

	render() {
		const {
			className,
			isIndeterminate,
			isDisabled,
			isSelected,
			onSelect,
			style,
			...passThroughs
		} = this.props;

		const labelChildProps = _.first(
			_.map(findTypes(this.props, CheckboxLabeled.Label), 'props')
		);

		return (
			<div className={cx('&-wrapper')}>
				<label
					className={cx(
						'&',
						{
							'&-is-disabled': isDisabled,
							'&-is-selected': isIndeterminate || isSelected,
						},
						className
					)}
					style={style}
				>
					<Checkbox
						className={cx('&-Checkbox', className)}
						isDisabled={isDisabled}
						isIndeterminate={isIndeterminate}
						isSelected={isSelected}
						onSelect={onSelect}
						{...omitProps(passThroughs, CheckboxLabeled, [], false)}
					/>
					<div
						{...labelChildProps}
						className={cx('&-label', _.get(labelChildProps, 'className', null))}
						onClick={_.noop}
					/>
				</label>
			</div>
		);
	},
});

export default CheckboxLabeled;
