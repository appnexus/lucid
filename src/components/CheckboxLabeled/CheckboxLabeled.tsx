import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import { findTypes, StandardProps } from '../../util/component-types';
import Checkbox, {
	ICheckboxProps,
	defaultProps as defaultPropsCheckbox,
} from '../Checkbox/Checkbox';

const cx = lucidClassNames.bind('&-CheckboxLabeled');
const { any, node, object, string, bool, func } = PropTypes;

export interface ILabelProps extends StandardProps {}

const Label = (props: ILabelProps): null => null;

Label.displayName = 'CheckboxLabeled.Label';
Label.peek = {
	description: `Renders a \`Label\` for the \`Checkbox\`.`,
	categories: ['controls', 'toggles'],
	madeFrom: ['Checkbox'],
};
Label.propName = 'Label';
Label.propTypes = {
	/**
		Used to identify the purpose of this checkbox to the user -- can be
		any renderable content.
	*/
	children: node,
};

/** TODO: Remove nonPassThroughs when the component is converted to a functional component */
const nonPassThroughs = [
	'isIndeterminate',
	'isDisabled',
	'isSelected',
	'onSelect',
	'className',
	'style',
	'Label',
];

export interface ICheckboxLabeledProps extends ICheckboxProps {
	/** Child element whose children are used to identify the purpose of this  checkbox to the user. */
	Label?: string | (React.ReactNode & { props: ILabelProps }) | React.ReactNode;
}

export const CheckboxLabeled = (
	props: ICheckboxLabeledProps
): React.ReactElement => {
	const {
		className,
		isIndeterminate,
		isDisabled,
		isSelected,
		onSelect,
		style,
		...passThroughs
	} = props;

	const labelChildProps = _.first(
		_.map(findTypes(props, CheckboxLabeled.Label), 'props')
	);

	return (
		<label
			className={cx(
				'&',
				{
					'&-is-disabled': isDisabled,
					'&-is-selected': isIndeterminate || isSelected,
				},
				className
			)}
			style={style}
		>
			<Checkbox
				className={cx('&-Checkbox', className)}
				isDisabled={isDisabled}
				isIndeterminate={isIndeterminate}
				isSelected={isSelected}
				onSelect={onSelect}
				{..._.omit(passThroughs, nonPassThroughs)}
			/>
			<div
				{...labelChildProps}
				className={cx('&-label', _.get(labelChildProps, 'className', null))}
			/>
		</label>
	);
};

CheckboxLabeled.displayName = 'CheckboxLabeled';

CheckboxLabeled.peek = {
	description: `A square two-state toggle with a \`Label\`.`,
	notes: {
		overview: `
			A square two-state toggle with a label that explains the action or selection. This is a composite of \`Checkbox\` and the native \`label\` element.
		`,
		intendedUse: `
			Use checkboxes to allow users to select one or more items. Commonly used to select filters or settings. For interactions where users can only select one option, use \`RadioButtonLabeled\`.
		`,
		technicalRecommendations: `
			- Use the styles on the \`CheckboxLabeled\` parent container to ensure only the checkboxes and their labels are clickable.
			- Use the Selected state when a filter or setting will be applied.
			- Use the Unselected state when a filter or setting will not be applied.
			- Use the Indeterminate state for parent checkboxes where some of the child checkboxes are Selected and some are Unselected. For example, the master checkbox in the header row of the interactive table example in \`DataTable\`.
			- You can have the label as a child or a prop depending on the needs of your application. 
		`,
	},
	categories: ['controls', 'toggles'],
	madeFrom: ['Checkbox'],
};

CheckboxLabeled.defaultProps = defaultPropsCheckbox;

// Can't just `...Checkbox.propTypes` anymore because of the way we have to
// handle default props. They are duplicated here on purpose which is okay
// since in the future we'll be removing proptypes in favor is just typescript.
CheckboxLabeled.propTypes = {
	/**
		Indicates whether the component should appear in an "indeterminate" or
		"partially checked" state. This prop takes precedence over
		\`isSelected\`.
	*/
	isIndeterminate: bool,

	/**
		Indicates whether the component should appear and act disabled by having
		a "greyed out" palette and ignoring user interactions.
	*/
	isDisabled: bool,

	/**
		Indicates that the component is in the "selected" state when true and in
		the "unselected" state when false. This props is ignored if
		\`isIndeterminate\` is \`true\`.
	*/
	isSelected: bool,

	/**
		Called when the user clicks on the component or when they press the space
		key while the component is in focus.  Signature:
		\`(isSelected, { event, props }) => {}\`
	*/
	onSelect: func,

	/**
		Appended to the component-specific class names set on the root element.
	*/
	className: string,

	/**
		Passed through to the root element.
	*/
	style: object,

	/**
		Child element whose children are used to identify the purpose of this
		checkbox to the user.
	*/
	Label: any,
};

CheckboxLabeled.Label = Label;

export default CheckboxLabeled;
