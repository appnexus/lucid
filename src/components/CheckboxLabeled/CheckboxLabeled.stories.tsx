import React, { useState } from 'react';
import { Meta } from '@storybook/react';
import CheckboxLabeled from './CheckboxLabeled';

export default {
	title: 'Controls/CheckboxLabeled',
	component: CheckboxLabeled,
	subcomponents: { 'CheckboxLabeled.Label': CheckboxLabeled.Label },
	parameters: {
		notes: CheckboxLabeled.peek.description,
	},
} as Meta;

const Template: any = (args) => {
	const [selected, setSelected] = useState(args.isSelected || false);
	return (
		<CheckboxLabeled
			{...args}
			isSelected={selected}
			onSelect={() => setSelected(!selected)}
		/>
	);
};

export const Default = Template.bind({});
Default.args = {
	title: 'Default',
	Label: 'Default',
};

export const Disabled = Template.bind({});
Disabled.args = {
	title: 'Disabled',
	Label: 'Disabled',
	isDisabled: true,
};

export const Selected = Template.bind({});
Selected.args = {
	title: 'Selected',
	Label: 'Selected',
	isSelected: true,
};

export const DisabledSelected = Template.bind({});
DisabledSelected.args = {
	title: 'Disabled Selected',
	Label: 'Disabled Selected',
	isDisabled: true,
	isSelected: true,
};

export const LabelAsProp = Template.bind({});
LabelAsProp.args = {
	Label: 'Label as prop',
	title: 'Label as prop',
};

export const HTMLElement = Template.bind({});
HTMLElement.args = {
	Label: <span>HTML element</span>,
	title: 'HTML element',
};

export const TextInAnArray = Template.bind({});
TextInAnArray.args = {
	Label: [
		'Text in an array',
		'Only the first value in the array is used',
		'The rest of these should be ignored',
	],
	title: 'Text in an array',
};

export const HTMLElementInAnArray = Template.bind({});
HTMLElementInAnArray.args = {
	Label: [
		<span key='1'>HTML element in an array</span>,
		<span key='2'>Again only the first value in the array is used</span>,
		<span key='3'>The rest should not be rendered</span>,
	],
	title: 'HTML element in an array',
};

export const LabelAsChild = (args) => {
	const [selected, setSelected] = useState(args.isSelected || false);

	return (
		<CheckboxLabeled
			{...args}
			isSelected={selected}
			onSelect={() => setSelected(!selected)}
		>
			<CheckboxLabeled.Label>
				<span>HTML element as Child</span>
			</CheckboxLabeled.Label>
		</CheckboxLabeled>
	);
};
