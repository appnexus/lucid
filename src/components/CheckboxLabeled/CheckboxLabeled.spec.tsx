import _ from 'lodash';
import assert from 'assert';
import React from 'react';
import { shallow } from 'enzyme';

import { common } from '../../util/generic-tests';

import CheckboxLabeled from './CheckboxLabeled';
import Checkbox from '../Checkbox/Checkbox';

describe('CheckboxLabeled', () => {
	common(CheckboxLabeled);

	describe('props', () => {
		describe('isDisabled', () => {
			it('passes the value through to its `Checkbox` instance.', () => {
				const wrapper = shallow(<CheckboxLabeled isDisabled={true} />);

				assert.equal(wrapper.find(Checkbox).prop('isDisabled'), true);
			});
		});

		describe('isSelected', () => {
			it('passes the value through to its `Checkbox` instance.', () => {
				const wrapper = shallow(<CheckboxLabeled isSelected={true} />);

				assert.equal(wrapper.find(Checkbox).prop('isSelected'), true);
			});
		});

		describe('onSelect', () => {
			it('passes the value through to its `Checkbox` instance.', () => {
				const foo = () => null;
				const wrapper = shallow(<CheckboxLabeled onSelect={foo} />);

				assert.equal(wrapper.find(Checkbox).prop('onSelect'), foo);
			});
		});

		describe('pass throughs', () => {
			it('passes through all props not defined in `propTypes` to its `Checkbox` instance.', () => {
				const wrapper = shallow(
					<CheckboxLabeled
						className='wut'
						isDisabled={true}
						isSelected={true}
						style={{ fontWeight: 'bold' }}
						onSelect={_.noop}
						foo={1}
						bar={2}
						baz={3}
						qux={4}
						quux={5}
					/>
				);
				const checkboxProps = wrapper.find(Checkbox).props();

				// It should pass `foo`, `bar`, `baz`, `qux`, and `quux` through
				// to the `Checkbox` instance.
				_.forEach(['foo', 'bar', 'baz', 'qux', 'quux'], prop => {
					assert(_.has(checkboxProps, prop));
				});
			});
		});
	});
});
