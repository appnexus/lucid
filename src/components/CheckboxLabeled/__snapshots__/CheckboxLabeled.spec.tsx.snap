// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 0.interactive 1`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    name="interactive-checkboxes"
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Vanilla
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Vanilla
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 0.interactive 2`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    name="interactive-checkboxes"
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Chocolate
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Chocolate
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 0.interactive 3`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    name="interactive-checkboxes"
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Strawberry
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Strawberry
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 0.interactive 4`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    name="interactive-checkboxes"
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Salted Caramel
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Salted Caramel
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 0.interactive 5`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    name="interactive-checkboxes"
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Mint chocolate chip (the best)
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Mint chocolate chip (the best)
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 1.states 1`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      (default props)
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    (default props)
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 1.states 2`] = `
<label
  className="lucid-CheckboxLabeled lucid-CheckboxLabeled-is-disabled"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={true}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Disabled
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Disabled
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 1.states 3`] = `
<label
  className="lucid-CheckboxLabeled lucid-CheckboxLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={true}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Selected
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Selected
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 1.states 4`] = `
<label
  className="lucid-CheckboxLabeled lucid-CheckboxLabeled-is-disabled lucid-CheckboxLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={true}
    isIndeterminate={false}
    isSelected={true}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Disabled & selected
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Disabled & selected
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 2.label-as-child 1`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      Just text
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Just text
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 2.label-as-child 2`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <CheckboxLabeled.Label>
      <span>
        HTML element
      </span>
    </CheckboxLabeled.Label>
  </Checkbox>
  <div
    className="lucid-CheckboxLabeled-label"
  >
    <span>
      HTML element
    </span>
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 3.label-as-prop 1`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Just text
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 3.label-as-prop 2`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <div
    className="lucid-CheckboxLabeled-label"
  >
    <span>
      HTML element
    </span>
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 3.label-as-prop 3`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <div
    className="lucid-CheckboxLabeled-label"
  >
    Text in an array
  </div>
</label>
`;

exports[`CheckboxLabeled [common] example testing should match snapshot(s) for 3.label-as-prop 4`] = `
<label
  className="lucid-CheckboxLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Checkbox
    className="lucid-CheckboxLabeled-Checkbox"
    isDisabled={false}
    isIndeterminate={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <div
    className="lucid-CheckboxLabeled-label"
  >
    <span
      key="1"
    >
      HTML element in an array
    </span>
  </div>
</label>
`;
